### DISCUSSION TITLE
Add more attributes?

### DISCUSSION THREAD

**Discussion by quack3d** [Chouffy/home_assistant_tgtg#19] [Ideas]
Are you able to fetch more data into the sensors? For instance price, URL to the item and pickup start/end like in this addon? https://community.home-assistant.io/t/toogoodtogo-mqtt-bridge/266197/23

**Comment by Chouffy** [comment]
Hi @sepodele , thanks for the suggestion!
That shouldn't be too hard - I'll have a look next week :)
If I remember correctly, the upstream tgtg python module already gives those info so I will just need to map it to Home Assistant

**Reply by quack3d** [reply]
Sweet. :) Would be nice to know if there are other interesting stuff in the upstream module as well.

**Reply by Chouffy** [reply]
You can check here in "Example response" https://github.com/ahivert/tgtg-python#get-an-item
Could you tell me which one are relevant for you?

**Reply by quack3d** [reply]
I hadn't seen the 'Example response' button. 🤭
It would be nice with the price (perhaps the price with taxes included?), the distance to the shop and URL to the item. I guess the URL to the item must be made by concatenating http://share.toogoodtogo.com/item/ and item_id. 

I would also like the pickup start/end but I don't see traces of that in the example? The addon linked above provides it so either the example is outdated or the addon don't use tgtg-python.

Fun to have: percentage and user_count under badges to show how well liked the item/store is.

**Reply by Chouffy** [reply]
Hey @sepodele I had some time around so I implemented some attributes.
You can use the [fresh 4.0.0 version ](https://github.com/Chouffy/home_assistant_tgtg/releases/tag/4.0.0) to use them! 

Thanks for the URL tip, if you didn't knew I wouldn't had the idea :)
Probably the tgtg-python example was outdated, so that's why you didn't saw the pickup start/end date but now it's here!

I didn't implemented your fun to have, but if you take a look at the [sensor.py](https://github.com/Chouffy/home_assistant_tgtg/blob/main/custom_components/tgtg/sensor.py), you can easily add your own in `extra_state_attributes`.
I probably won't have the time to work on it more, but you're way more than welcome to create a Pull Request so everybody can benefit!

**Reply by quack3d** [reply]
Brilliant! Works like a charm! Thank you! :)
